pkg/influxunifi/integration_test.go

```diff
@@ -0,0 +1,173 @@
+package influxunifi
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"testing"
+	"time"
+
+	influxV1Models "github.com/influxdata/influxdb1-client/models"
+	influxV1 "github.com/influxdata/influxdb1-client/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/unpoller/unpoller/pkg/testutil"
+)
+
+type mockInfluxV1Client struct {
+	databases map[string]bool
+	points    map[string]foundPoint
+}
+
+type foundPoint struct {
+	Tags   map[string]string `json:"tags"`
+	Fields map[string]string `json:"fields"`
+}
+
+func newMockInfluxV1Client() *mockInfluxV1Client {
+	return &mockInfluxV1Client{
+		databases: make(map[string]bool, 0),
+		points:    make(map[string]foundPoint, 0),
+	}
+}
+
+func (m *mockInfluxV1Client) Ping(timeout time.Duration) (time.Duration, string, error) {
+	return time.Millisecond, "", nil
+}
+
+func influxV1FieldTypeToString(ft influxV1Models.FieldType) string {
+	switch ft {
+	case influxV1Models.Integer:
+		return "int"
+	case influxV1Models.Float:
+		return "float"
+	case influxV1Models.Boolean:
+		return "bool"
+	case influxV1Models.Empty:
+		return "none"
+	case influxV1Models.String:
+		return "string"
+	case influxV1Models.Unsigned:
+		return "uint"
+	default:
+		return "unknown"
+	}
+}
+
+func (m *mockInfluxV1Client) Write(bp influxV1.BatchPoints) error {
+	m.databases[bp.Database()] = true
+	for _, p := range bp.Points() {
+		m.points[p.Name()] = foundPoint{
+			Tags:   p.Tags(),
+			Fields: make(map[string]string),
+		}
+		fields, err := p.Fields()
+		if err != nil {
+			continue
+		}
+		point, _ := influxV1Models.NewPoint(p.Name(), influxV1Models.NewTags(p.Tags()), fields, p.Time())
+		fieldIter := point.FieldIterator()
+		for fieldIter.Next() {
+			fieldName := string(fieldIter.FieldKey())
+			fieldType := influxV1FieldTypeToString(fieldIter.Type())
+			m.points[p.Name()].Fields[fieldName] = fieldType
+		}
+	}
+	return nil
+}
+
+func (m *mockInfluxV1Client) Query(q influxV1.Query) (*influxV1.Response, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+func (m *mockInfluxV1Client) QueryAsChunk(q influxV1.Query) (*influxV1.ChunkedResponse, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+func (m *mockInfluxV1Client) Close() error {
+	return nil
+}
+
+type testPointExpectation struct {
+	Tags   []string          `json:"tags"`
+	Fields map[string]string `json:"fields"`
+}
+
+type testExpectations struct {
+	Databases []string                        `json:"databases"`
+	Points    map[string]testPointExpectation `json:"points"`
+}
+
+//go:embed integration_test_expectations.json
```

Comment:
Embedding a file for a test that you can't (don't) run while compiled seems suspicious. Maybe this is normal? Convince me. :)

Reply:
this is a good reminder to clean that up, YAML would be a better alternative here.

To the comment below here as well, I will probably take the dump command here as an extension to also generate these as well in a follow up PR

Reply:
cleaned up